### ISSUE TITLE
about the decrease of the loss

### ISSUE THREAD

**Issue by hhxzzy** [tfzhou/ContrastiveSeg#44]
Hello. Thanks for your excellent work!
I transplant the loss function and memory bank of your code to my own code, and run my code on Cityscapes dataset and my own dataset. But it turns out that the decrease of the contrast loss is very small. For example, the contrast loss drops from 1.27 to 1.11 after 80k epoch on Cityscapes, while the ce loss drops from 1.26 to 0.15. And the same thing happens on my own dataset. It seems that the contrast loss is not being useful during train.
I wonder is it normal for the contrast loss to decrease so little? And what can I do to make full use of the contrast loss, like tuning the hyper-parameters?

**Comment by guwen007** [comment]
I have the same problem too, looking forward to the author's answer

**Comment by wingkeihon** [comment]
@hhxzzy I'm sorry to bother you.I added the contrast loss function to my model and code, and this problem also occurred.Has your problem been solved?

**Comment by hhxzzy** [comment]
> @hhxzzy I'm sorry to bother you.I added the contrast loss function to my model and code, and this problem also occurred.Has your problem been solved?

Hello. I try to make some change to the code, but the contrastive loss is still not going to drop much. But it seems that adding the contrastive can help improve the mIOU on Cityscapes by about 1%. Maybe it is the nature of contrastive loss that it doesn't drop much.
By the way, do you use warmup_iter in your code. I found that using a warm up strategy can be helpful. The author use this strategy in his code but not metion it in the paper.

**Comment by wingkeihon** [comment]
已收到

**Comment by Pantherk1** [comment]
@hhxzzy 
I am a novice postgraduate, and my coding ability is not high. I want to use ContrastiveSeg method on my own two-class segmentation model. Which py files should I read or what should I pay attention to?
Looking forward to your answer, thank you very much.

**Comment by Pantherk1** [comment]
@tfzhou @hhxzzy @wingkeihon @guwen007 
I am a novice postgraduate, and my coding ability is not high. I want to use ContrastiveSeg method on my own two-class segmentation model. Which py files should I read or what should I pay attention to?
Looking forward to your answer, thank you very much.

**Comment by Pantherk1** [comment]
> > @tfzhou @hhxzzy @wingkeihon @guwen007 I am a novice postgraduate, and my coding ability is not high. I want to use ContrastiveSeg method on my own two-class segmentation model. Which py files should I read or what should I pay attention to? Looking forward to your answer, thank you very much.
> 
> If you want to run the author's scripts directly, you can take configs/cityscapes/H_48_D_4.json as an example. You need to change "data" and "loss“/"params"/"ce_weight" according to your dataset. If you wanna use contrastive loss on your own code instead of running the author's scripts, you should look at lib/loss/loss_contrast.py and lib/loss/loss_contrast_mem.py to read the implement of contrastive loss.

Thank you for your prompt attention.
